File: firefly-example/src/test/java/com/firefly/example/test/dao/TestUserDAO.java
package com.firefly.example.test.dao;

import com.firefly.example.reactive.coffee.store.dao.UserDAO;
import com.firefly.example.test.TestBase;
import org.junit.Assert;
import org.junit.Test;
import reactor.test.StepVerifier;

import static com.firefly.example.reactive.coffee.store.AppMain.javaCtx;
import static org.hamcrest.Matchers.is;

/**
 * @author Pengtao Qiu
 */
public class TestUserDAO extends TestBase {

    private UserDAO userDAO = javaCtx.getBean(UserDAO.class);

    @Test
    public void test() {
        StepVerifier.create(userDAO.getByName("John")).assertNext(user -> {
            Assert.assertThat(user.getId(), is(2L));
            Assert.assertThat(user.getPassword(), is("123456"));
            System.out.println(user);
        }).expectComplete().verify();
    }
}

File: firefly-example/src/test/kotlin/com/firefly/example/kotlin/test/dao/TestUserDAO.kt
package com.firefly.example.kotlin.test.dao

import com.firefly.example.kotlin.coffee.store.dao.UserDAO
import com.firefly.example.kotlin.coffee.store.ktCtx
import com.firefly.example.kotlin.test.TestBase
import com.firefly.kotlin.ext.context.getBean
import kotlinx.coroutines.experimental.runBlocking
import org.junit.Test
import kotlin.test.assertEquals

/**
 * @author Pengtao Qiu
 */
class TestUserDAO : TestBase() {

    private val userDAO = ktCtx.getBean<UserDAO>()

    @Test
    fun test() = runBlocking {
        val user = userDAO.getByName("John")
        assertEquals(2L, user.id)
        assertEquals("123456", user.password)
    }
}